src/_P009_MCP.ino

```diff
@@ -67,31 +124,151 @@ boolean Plugin_009(byte function, struct EventStruct *event, String& string)
         Plugin_009_Config(Settings.TaskDevicePort[event->TaskIndex], 1);
 
         // read and store current state to prevent switching at boot time
-        switchstate[event->TaskIndex] = Plugin_009_Read(Settings.TaskDevicePort[event->TaskIndex]);
+        switchstate[event->TaskIndex] = (int) Plugin_009_Read(Settings.TaskDevicePort[event->TaskIndex]);
+
+        // @giig1967g-20181022: set initial UserVar of the switch
+        if (Settings.TaskDevicePin1Inversed[event->TaskIndex]){
+          UserVar[event->BaseVarIndex] = !switchstate[event->TaskIndex];
+        } else {
+          UserVar[event->BaseVarIndex] = switchstate[event->TaskIndex];
+        }
 
         // if boot state must be send, inverse default state
         if (Settings.TaskDevicePluginConfig[event->TaskIndex][0])
           switchstate[event->TaskIndex] = !switchstate[event->TaskIndex];
 
+        // @giig1967g-20181022: doubleclick counter = 0
+        Settings.TaskDevicePluginConfig[event->TaskIndex][7]=0;
+
+        // @giig1967g-20181022: used to track if LP has fired
+        Settings.TaskDevicePluginConfig[event->TaskIndex][6]=false;
+
+        // @giig1967g-20181022: store millis for debounce, doubleclick and long press
+        Settings.TaskDevicePluginConfigLong[event->TaskIndex][0]=millis(); //debounce timer
+        Settings.TaskDevicePluginConfigLong[event->TaskIndex][1]=millis(); //doubleclick timer
+        Settings.TaskDevicePluginConfigLong[event->TaskIndex][2]=millis(); //longpress timer
+
+        // @giig1967g-20181022: set minimum value for doubleclick MIN max speed
+        if (Settings.TaskDevicePluginConfigFloat[event->TaskIndex][1] < PLUGIN_009_DOUBLECLICK_MIN_INTERVAL)
+          Settings.TaskDevicePluginConfigFloat[event->TaskIndex][1] = PLUGIN_009_DOUBLECLICK_MIN_INTERVAL;
+
+        // @giig1967g-20181022: set minimum value for longpress MIN max speed
+        if (Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2] < PLUGIN_009_LONGPRESS_MIN_INTERVAL)
+          Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2] = PLUGIN_009_LONGPRESS_MIN_INTERVAL;
+
+        // @giig1967g-20181022: set initial UserVar of the switch
+        UserVar[event->BaseVarIndex] = switchstate[event->TaskIndex];
```

Comment:
This was set in line 131 or 133.
Maybe set bootstate (line 137/138) before setting the Uservar values?

Current logic is too complicated to oversee all possibilities of pin state inverse and boot state.

Reply:
will review it

Reply:
> This was set in line 131 or 133.
Right. Fixed.

> Maybe set bootstate (line 137/138) before setting the Uservar values?
Will do that. But it won't matter, as immidiately after PLUGIN_TEN_PER_SECOND will overwrite switchstate and UserVar anyway.

> Current logic is too complicated to oversee all possibilities of pin state inverse and boot state.
I have reviewed the whole logic and will post it soon